externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/SparkSQLEngine.scala

```diff
@@ -54,7 +54,12 @@ case class SparkSQLEngine(
     super.start()
     // Start engine self-terminating checker after all services are ready and it can be reached by
     // all servers in engine spaces.
-    backendService.sessionManager.startTerminatingChecker()
+    backendService.sessionManager.startTerminatingChecker(() => {
+      currentEngine match {
```

Comment:
if we go into here that means the `currentEngine` has already initialized. I think we can do:
```java
assert(currentEngine.isDefined)
currentEngine.get.stop()
```